tests/evm/test_log.py

```diff
@@ -0,0 +1,187 @@
+import pytest
+
+from zkevm_specs.evm import (
+    ExecutionState,
+    StepState,
+    verify_steps,
+    Tables,
+    RWTableTag,
+    TxLogFieldTag,
+    CallContextFieldTag,
+    RW,
+    Block,
+    Bytecode,
+    LOG_STATIC_GAS,
+)
+from zkevm_specs.util import rand_range, rand_address, rand_fp, RLC, U64, U256, memory_expansion
+
+# TESTING_DATA = (topics, mstart, msize)
+TESTING_DATA = (
+    # zero topic
+    ([], 10, 2),
+    # one topic
+    ([0x030201], 20, 3),
+    # two topics
+    ([0x030201, 0x0F0E0D], 100, 20),
+    # four topics
+    ([0x030201, 0x0F0E0D, 0x0D8F01, 0x0AA213], 421, 15),
+)
+CALLEE_ADDRESS = rand_address()
+
+bytecodes = [Bytecode().log0(), Bytecode().log1(), Bytecode().log2(), Bytecode().log3(), Bytecode().log4()]
+
+
+@pytest.mark.parametrize("topics, mstart, msize", TESTING_DATA)
+def test_log(topics: list, mstart: U64, msize: U64):
+    randomness = rand_fp()
+    block = Block()
+
+    data = []
+    for i in range(msize):
+        data.append(rand_range(255))
+
+    topic_count = len(topics)
+    next_memory_size, memory_expansion_cost = memory_expansion(mstart, msize)
+    dynamic_gas = LOG_STATIC_GAS * topic_count + 8 * msize + memory_expansion_cost
+
+    bytecode = bytecodes[topic_count]
+    bytecode_hash = RLC(bytecode.hash(), randomness)
+
+    tables = Tables(
+        block_table=set(block.table_assignments(randomness)),
+        tx_table=set(),
+        bytecode_table=set(bytecode.table_assignments(randomness)),
+        rw_table=set(
+            [
+                (1, RW.Read, RWTableTag.Stack, 1, 1017, 0, RLC(mstart, randomness, 8), 0, 0, 0),
+                (2, RW.Read, RWTableTag.Stack, 1, 1018, 0, RLC(msize, randomness, 8), 0, 0, 0),
+            ]  # write topics
+            + construct_topic_stack(3, 1019, topics, randomness)
+            + construct_memory_rows(3 + topic_count, data, mstart, randomness)
+            + construct_topic_rows(3 + topic_count + msize, topics, randomness)
+            + construct_data_rows(3 + 2 * topic_count + msize, data, randomness)
+            + [
+                # demo rows of one topic with two data
+                # (3, RW.Read, RWTableTag.Stack, 1, 1022, 0, RLC(topic1, randomness, 32), 0, 0, 0),
+                # (4, RW.Read, RWTableTag.Memory, 1, 11, 0, 10, 0, 0, 0),
+                # (5, RW.Read, RWTableTag.Memory, 1, 12, 0, 20, 0, 0, 0),
+                # write tx log with topic and data
+                # (6, RW.Write, RWTableTag.TxLog, 0, 0, TxLogFieldTag.Topics, RLC(topic1, randomness, 32), 0, 0, 0),
+                # (7, RW.Write, RWTableTag.TxLog, 0, 0, TxLogFieldTag.Data, 10, 0, 0, 0),
+                # (8, RW.Write, RWTableTag.TxLog, 0, 1, TxLogFieldTag.Data, 20, 0, 0, 0),
```

Comment:
Are these intended to use or should they be removed?

Reply:
it is not using but for illustrating what's the `RW` table look like after adding `TxLog`.  let me know if you think it is non sense

Reply:
Keeping them seems fine 👍